src/conditionalAndLoops.js

```diff
@@ -19,160 +19,306 @@
+		return "a month ago";
+	  }
+	
+	  if (timeDiff <= 345 * oneDay) {
+		const months = Math.round((timeDiff - 1) / oneMonth);
+		return `${months} months ago`;
+	  }
+	
+	  if (timeDiff <= 545 * oneDay) {
+		return "a year ago";
+	  }
+	
+	  const years = Math.round((timeDiff - 1) / oneYear);
+	  return `${years} years ago`;
+	}
+	
+  //Super Super Super Hard
+  // function timespanToHumanString(startTime, endTime) {
+  //   const differenceInSeconds = Math.floor((endTime - startTime) / 1000);
+  //   const differenceInMinutes = Math.floor(differenceInSeconds / 60);
+  //   const differenceInHours = Math.floor(differenceInMinutes / 60);
+  //   const differenceInDays = Math.floor(differenceInHours / 24);
+  //   const differenceInMonths = Math.floor(differenceInDays / 30);
+  //   const differenceInYears = Math.floor(differenceInMonths / 12);
+  
+  //   if (differenceInSeconds == 45) {
+  //     return "a few seconds ago";
+  //   } else if (differenceInSeconds > 45 && differenceInSeconds <= 90) {
+  //     return "a minute ago";
+  //   } else if (Math.ceil(differenceInSeconds) <= 45) {
+  //     return "a few seconds ago";
+  //   } else if (differenceInSeconds > 90 && differenceInSeconds <= 120) {
+  //     return `2 minutes ago`;
+  //   } else if (differenceInMinutes <= 45) {
+  //     return `${Math.floor(differenceInMinutes)} minutes ago`;
+  //   } else if (differenceInMinutes >= 45 && differenceInMinutes <= 90) {
+  //     return "an hour ago";
+  //   } else if (differenceInHours <= 22) {
+  //     return `${Math.ceil(differenceInMinutes) / 60} hours ago`;
+  //   } else if (differenceInHours > 22 && differenceInHours <= 24) {
+  //     return "a day ago";
+  //   } else if (differenceInDays <= 25) {
+  //     return `${Math.ceil(differenceInDays)} days ago`;
+  //   } else if (differenceInDays > 25 && differenceInDays <= 45) {
+  //     return "a month ago";
+  //   } else if (differenceInDays >= 45 && differenceInDays <= 345) {
+  //     return `${Math.ceil(differenceInDays / 30)} months ago`;
+  //   } else if (differenceInDays >= 345 && differenceInDays <= 545) {
+  //     return "a year ago";
+  //   } else {
+  //     return `${Math.ceil(differenceInDays / 30)} years ago`;
+  //   }
+  // }
+  
+  
+  /**
+   * Returns the string with n-ary (binary, ternary, etc, where n<=10) representation of
+   * specified number.
+   * See more about
+   * https://en.wikipedia.org/wiki/Binary_number
+   * https://en.wikipedia.org/wiki/Ternary_numeral_system
+   * https://en.wikipedia.org/wiki/Radix
+   *
+   * @param {number} num
+   * @param {number} n, radix of the result
+   * @return {string}
+   *
+   * @example:
+   *   1024, 2  => '10000000000'
+   *   6561, 3  => '100000000'
+   *    365, 2  => '101101101'
+   *    365, 3  => '111112'
+   *    365, 4  => '11231'
+   *    365, 10 => '365'
+   */
+  function toNaryString(num, n) {
+	if (n < 2 || n > 10) {
+	  throw new Error("Radix must be between 2 and 10");
+	}
```

Comment:
As a challenge, try to change this function so that it works for any bases till 16.